### ISSUE TITLE
sigmoid operating on tensors only

### ISSUE THREAD

**Issue by tuananhle7** [twitter/torch-autograd#115]
Running
```
grad = require 'autograd'
params = {w = 1}
f = function(params, x)
   return torch.sigmoid(params.w * x)
end
df = grad(f, {optimize = true})
dparams, loss = df(params, 2)
```
produces
```
/Users/tuananhle/torch/install/bin/luajit: invalid arguments: number number
expected arguments: [*DoubleTensor*] DoubleTensor DoubleTensor
stack traceback:
	[C]: at 0x0fc48540
	[C]: in function 'fn'
	.../install/share/lua/5.1/autograd/runtime/codegen/Node.lua:72: in function 'evaluateForward'
	...install/share/lua/5.1/autograd/runtime/codegen/Graph.lua:25: in function 'cmul'
	...nanhle/torch/install/share/lua/5.1/autograd/gradfuns.lua:554: in function <...nanhle/torch/install/share/lua/5.1/autograd/gradfuns.lua:553>
	.../install/share/lua/5.1/autograd/runtime/codegen/Node.lua:127: in function 'evaluateBackward'
	...install/share/lua/5.1/autograd/runtime/codegen/Graph.lua:370: in function 'protectedFn'
	...install/share/lua/5.1/autograd/runtime/codegen/Graph.lua:383: in function 'record'
	.../install/share/lua/5.1/autograd/runtime/codegen/init.lua:44: in function 'generateFn'
	.../install/share/lua/5.1/autograd/runtime/codegen/init.lua:140: in function 'df'
	test.lua:14: in main chunk
	[C]: in function 'dofile'
	...nhle/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
	[C]: at 0x010f7debc0
```
pointing to an error at [gradfuns.lua:554](https://github.com/twitter/torch-autograd/blob/master/src/gradfuns.lua#L553), while 
```
grad = require 'autograd'
params = {w = torch.Tensor{1}}
f = function(params, x)
    return torch.sigmoid(torch.cmul(params.w, x))[1]
end
df = grad(f, {optimize = true})
dparams, loss = df(params, torch.Tensor{2})
```
works fine.

**Comment by alexbw** [comment]
Does `torch.sigmoid` work with numbers, outside of autograd? If not, you'll need to upgrade your version of torch.

**Comment by tuananhle7** [comment]
Yes, `torch.sigmoid` works fine for me outside of autograd.

**Comment by alexbw** [comment]
Should be fixed in 14e7f9b
